src/en/news/blog/2022/improving-dashboard-accessibility-for-blind-and-visually-impaired-ceph-users/index.md

```diff
@@ -0,0 +1,134 @@
+
+- Added accessible labels for row selection checkboxes for better identification by screen readers
+- Added a custom pagination component with better UX and accessibility
+- Changed table action button label colors from red to cyan for better legibility and focus
+
+![Datatable](images/datatable.png)
+
+Many more sub-components were also updated and can be found on [GitHub](https://github.com/ceph/ceph/pulls/nsedrickm)
+
+### Accessibility testing
+
+To prevent future regressions, we added end to end accessibility tests for high impact components like the dashboard and navigation. If any future regressions on these components occurs, a tabulated error description is generated and added to the Cypress log for debugging.
+
+![Error message](images/error.png)
+
+More information on accessibility e2e testing can be found in the dashboard developer [documentation](https://docs.ceph.com/en/latest/dev/developer_guide/dash-devel/#accessibility).
+
+## Challenges faced
+
+While working on this project, I ran into several challenges. Actually tracking down the code where fixes needed to be made was quite a task most times, given how large the Ceph codebase is. Using `grep` helped me find code sections a lot faster.
+
+Also, some components like the datatable are actually provided by a third party library. This made it a bit challenging to improve on the accessibility of certain parts. Fortunately, we could develop and use custom templates instead with better accessibility. In some extreme cases whole components had to be replaced like the table pagination component.
+
+In addition some updates I made altered the QA tests currently in place for the Dashboard. This earned me several days of debugging to figure out what I was doing wrong and how to seek help when stuck.
+
+## Conclusion
+
+Accessibility is quite important and should be considered for every application used by a wide audience. Though at the time of this writing all the Pull requests have not been merged, we can be certain the next official release of Ceph dashboard will be more WCAG 2.1 level A compliant and therefore improved for blind and visually impaired Ceph users.
+
+This was truly an engaging experience for me, I learned a lot while also contributing substantially to Ceph. I would like to thank my mentors Laura Flores and Ernesto Puerta and the entire Dashboard team for all their relentless support and guidance.
+
+Not only have I picked up new technical skills like Cypress and e2e testing but I have also learned how to communicate, contribute to discussions and also seek help when stuck on challenges.
+
+## Pending updates
+
+While we aimed to also add a dark mode feature to the dashboard during this period we unfortunately could not due to time constraints. Not to worry, it sure is planned and will be added soon. I remain a ceph contributor and will continue to make improvements wherever I can
```

Comment:
No need to write this section. Whenever the dark mode feature is introduced, we can write another blog post about it!

Reply:
I felt the need to add it here because it was one of the deliverables promised and the gsoc f[inal submission requirements](https://developers.google.com/open-source/gsoc/help/work-product)  says to mention any work left

Reply:
@nSedrickm sure, makes sense. In that case, I have one final edit for you. Otherwise, everything looks good:

Grammar and "in the future", so you're not boxing yourself into more time constraints.
```suggestion
## Pending updates

While we aimed to also add a dark mode feature to the dashboard during this period, we unfortunately could not due to time constraints. Not to worry, it sure is planned and will be added in the future. I remain a Ceph contributor and will continue to make improvements wherever I can.
```